lifecycle/lifecycle/deployer/provision.py

```diff
@@ -57,15 +57,18 @@ def provision_job(
         family = create_job_family_if_not_exist(manifest.name)
         family_dto = job_family_model_to_dto(family)
 
-        runtime_env_vars = merge_env_vars(manifest.runtime_env, secret_runtime_env)
         build_env_vars = merge_env_vars(manifest.build_env, secret_build_env)
-        runtime_env_vars = hide_env_vars(runtime_env_vars, build_env_vars)
+        all_runtime_vars = merge_env_vars(manifest.runtime_env, secret_runtime_env)
+        runtime_env_vars = manifest.runtime_env or {}
+        for build_var in build_env_vars.keys():
+            if build_var not in all_runtime_vars:
+                runtime_env_vars[build_var] = ''  # Clear out unused build env vars
```

Comment:
This comment would be much more useful if instead of what is happening (it's visible), it explained why it is needed.